### DISCUSSION TITLE
app.openapi() schema silently breaks when something which is "unschemafiable" is encountered in a response model

### DISCUSSION THREAD

**Discussion by zakstucke** [fastapi/fastapi#10003] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import typing as tp
from fastapi import FastAPI
from pydantic import BaseModel, Field

app = FastAPI()

class Foo(BaseModel):
    bar: int
    handler: tp.Callable[[], None] = Field(exclude=True)

@app.post("/check")
def validator_check() -> Foo:
    return Foo(handler=lambda: None, bar=1)

app.openapi()
```

### Description

As discussed in https://github.com/tiangolo/fastapi/discussions/9709#discussioncomment-6435347 (and https://github.com/pydantic/pydantic/issues/6630)

When an "unschemafiable" field is encountered schema breaks, even though it has `Field(exclude=True)`.

With `handler` property, MRE outputs: (broken)
```JSON
{
   "openapi":"3.1.0",
   "info":{
      "title":"FastAPI",
      "version":"0.1.0"
   },
   "paths":{
      "/check":{
         "post":{
            "summary":"Validator Check",
            "operationId":"validator_check_check_post",
            "responses":{
               "200":{
                  "description":"Successful Response",
                  "content":{
                     "application/json":{
                        "schema":{
                           "$ref":"#/components/schemas/__main____FooInput__1"
                        }
                     }
                  }
               }
            }
         }
      }
   }
}
```

When `handler` property removed MRE outputs: (fixed)
```JSON
{
   "openapi":"3.1.0",
   "info":{
      "title":"FastAPI",
      "version":"0.1.0"
   },
   "paths":{
      "/check":{
         "post":{
            "summary":"Validator Check",
            "operationId":"validator_check_check_post",
            "responses":{
               "200":{
                  "description":"Successful Response",
                  "content":{
                     "application/json":{
                        "schema":{
                           "$ref":"#/components/schemas/Foo"
                        }
                     }
                  }
               }
            }
         }
      }
   },
   "components":{
      "schemas":{
         "Foo":{
            "properties":{
               "bar":{
                  "type":"integer",
                  "title":"Bar"
               }
            },
            "type":"object",
            "required":[
               "bar"
            ],
            "title":"Foo"
         }
      }
   }
}
```

### Operating System

Linux

### FastAPI Version

0.100.1

### Pydantic Version

2.1.1

### Python Version

3.11.4

**Comment by zakstucke** [comment, marked as answer]
Marking as closed as this was fixed in `0.101.0`   :tada:

**Comment by verngutz** [comment]
Using the `separate_input_output_schemas=False` option introduced in FastAPI 0.102.0 causes the example to throw `pydantic.errors.PydanticInvalidForJsonSchema: Cannot generate a JsonSchema for core_schema.CallableSchema`.
